File: typegate/tests/runtimes/deno/deno_dep.py
from typegraph.graph.typegraph import Graph
from typegraph.policy import Policy
from typegraph.runtimes.deno import DenoRuntime

from typegraph import t, typegraph


@typegraph()
def deno_dep(g: Graph):
    deno = DenoRuntime()
    public = Policy.public()

    g.expose(
        public,
        doAddition=deno.import_(
            t.struct({"a": t.float(), "b": t.float()}),
            t.float(),
            module="ts/dep/main.ts",
            deps=["ts/dep/nested/dep.ts"],
            name="doAddition",
        ),
        simple=deno.func(
            t.struct({"a": t.float(), "b": t.float()}),
            t.float(),
            code="({ a, b }) => a + b",
        ),
    )

File: typegate/tests/runtimes/deno/deno_dep.ts
// Copyright Metatype OÜ, licensed under the Elastic License 2.0.
// SPDX-License-Identifier: Elastic-2.0

import { Policy, t, typegraph } from "@typegraph/sdk/index.js";
import { DenoRuntime } from "@typegraph/sdk/runtimes/deno.js";

export const tg = await typegraph({
  name: "test-deno-dep",
}, (g: any) => {
  const deno = DenoRuntime();
  const pub = Policy.public();

  g.expose({
    doAddition: deno.import(
      t.struct({ a: t.float(), b: t.float() }),
      t.float(),
      {
        module: "ts/dep/main.ts",
        name: "doAddition",
      },
    ).withPolicy(pub),
  });
});
